extensions/elytron-security/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronRecorder.java

```diff
@@ -81,7 +85,12 @@ public boolean implies(Permission permission) {
      * @param realm - the runtime value for the SecurityRealm
      */
     public void addRealm(RuntimeValue<SecurityDomain.Builder> builder, String realmName, RuntimeValue<SecurityRealm> realm) {
-        builder.getValue().addRealm(realmName, realm.getValue());
+        builder.getValue().addRealm(realmName, realm.getValue()).setRoleDecoder(new RoleDecoder() {
+            @Override
+            public Roles decodeRoles(AuthorizationIdentity authorizationIdentity) {
+                return CDI.current().select(DefaultRoleDecoder.class).get().decodeRoles(authorizationIdentity);
+            }
+        });
```

Comment:
Am I missing something or did you lose the call to `build()`? :)

Reply:
I am the one losing my head... sorry :sweat_smile: